GVRf/Sample/gvrf_immersivepedia/src/org/gearvrf/immersivepedia/scene/DinosaurScene.java

```diff
@@ -0,0 +1,183 @@
+    public DinosaurScene(GVRContext gvrContext) throws IOException {
+        super(gvrContext);
+        this.gvrContext = gvrContext;
+        DinosaurFactory.getInstance(gvrContext);
+        getMainCameraRig().getTransform().setPositionY(CAMERA_Y);
+
+        createVideoDinosauGroup(); // TRex
+        createTextDinosaurGroup(); // Ankylosaurus
+        createRotateDinosaurGroup(); // Styracosaurus
+        createGalleryDinosaurGroup(); // Apatosaurus
+
+        addSceneObject(createSkybox()); //
+
+        hide();
+        addSceneObject(createBlueSkybox()); //
+
+    }
+
+    private void createRotateDinosaurGroup() throws IOException {
+        RotateDinosaurGroup rotateDinosaur = new RotateDinosaurGroup(getGVRContext(), this);
+        addSceneObject(rotateDinosaur);
+    }
+
+    private void createTextDinosaurGroup() throws IOException {
+
+        textDinosaur = new TextDinosaurGroup(getGVRContext(), this);
+
+        textDinosaur.getTransform().setPositionZ(-DinosaurFactory.ANKYLOSAURUS_DISTANCE);
+
+        textDinosaur.getTransform().rotateByAxisWithPivot(
+                DinosaurFactory.ANKYLOSAURUS_ANGLE_AROUND_CAMERA, 0, 1, 0, 0, 0, 0);
+
+        addSceneObject(textDinosaur);
+
+    }
+
+    private void createVideoDinosauGroup() throws IOException {
+
+        videoDinosaur = new VideoDinosaurGroup(getGVRContext(), this);
+
+        videoDinosaur.getTransform().setPositionZ(-DinosaurFactory.TREX_DISTANCE);
+
+        videoDinosaur.getTransform().rotateByAxisWithPivot(
+                DinosaurFactory.TREX_ANGLE_AROUND_CAMERA, 0, 1, 0, 0, 0, 0);
+
+        addSceneObject(videoDinosaur);
+    }
+
+    private void createGalleryDinosaurGroup() throws IOException {
+        galleryDinosaur = new GalleryDinosaurGroup(gvrContext, this);
+        addSceneObject(galleryDinosaur);
+    }
+
+    public void onStep() {
+        if (this.videoDinosaur != null) {
+            this.videoDinosaur.onStep();
+        }
+    }
+
+    public void show() {
+        getGVRContext().setMainScene(this);
+        GazeController.enableGaze();
+        for (GVRSceneObject object : getWholeSceneObjects()) {
+            if (object.getRenderData() != null && object.getRenderData().getMaterial() != null) {
+                new GVROpacityAnimation(object, 1f, 1f).start(getGVRContext().getAnimationEngine());
+            }
+        }
+    }
+
+    public void hide() {
+        for (GVRSceneObject object : getWholeSceneObjects()) {
+            if (object.getRenderData() != null && object.getRenderData().getMaterial() != null) {
+                object.getRenderData().getMaterial().setOpacity(0f);
+            }
+        }
+    }
+
+    private GVRSceneObject createSkybox() {
+
+        GVRMesh mesh =  getGVRContext().loadMesh(new GVRAndroidResource(getGVRContext(), R.raw.environment_walls_mesh));
+        GVRTexture texture = getGVRContext().loadTexture(Loader.loadResourceFromFile(getGVRContext(),
+                "environment_walls_tex_diffuse.png", 0));
```

Comment:
This problem is present in few places. I don't have "environment_walls_tex_diffuse.png" in my externalStorageDirectory - should I? Otherwise this and few other textures do not load and the app bails out.